portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -199,6 +205,24 @@ void vPortStartFirstTask( void )
 }
 /*-----------------------------------------------------------*/
 
+#if ( configUSE_IDLE_HOOK == 1 )
+/** Called from the idle task */
+void vApplicationIdleHook()
+{
+    // call sleep() to provide a thread cancellation point, necessary
+    // for pthread_cancel() to be able to cancel the idle thread
+    sleep(0);
+}
+#else
+#error POSIX port requires configUSE_IDLE_HOOK be defined as 1
+#endif /* configUSE_IDLE_HOOK */
```

Comment:
Due to the double definition I believe this becomes a breaking change for all demos that already provided or used this function though?
It's currently causing a failure to build the existing [Posix_GCC demo](https://github.com/FreeRTOS/FreeRTOS/blob/main/FreeRTOS/Demo/Posix_GCC/main.c#L214) as seen [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/actions/runs/7041042651/job/19162957971#step:5:167)